packages/@aws-cdk/aws-codestarnotifications/test/rule.test.ts

```diff
@@ -0,0 +1,222 @@
+    expect(stack).toHaveResourceLike('AWS::CodeStarNotifications::NotificationRule', {
+      Name: 'MyNotificationRuleGeneratedFromId',
+      Resource: 'arn:aws:codebuild::1234567890:project/MyCodebuildProject',
+      Status: 'ENABLED',
+    });
+  });
+
+  test('generating name will cut if id length is over than 64 charts', () => {
+    const project = new FakeCodeBuild();
+
+    new notifications.NotificationRule(stack, 'MyNotificationRuleGeneratedFromIdIsToooooooooooooooooooooooooooooLong', {
+      source: project,
+    });
+
+    expect(stack).toHaveResourceLike('AWS::CodeStarNotifications::NotificationRule', {
+      Name: 'MyNotificationRuleGeneratedFromIooooooooooooooooooooooooooooLong',
+      Resource: 'arn:aws:codebuild::1234567890:project/MyCodebuildProject',
+      Status: 'ENABLED',
+    });
+  });
+
+  test('created new notification rule without detailType', () => {
+    const project = new FakeCodeBuild();
+
+    new notifications.NotificationRule(stack, 'MyNotificationRule', {
+      notificationRuleName: 'MyNotificationRule',
+      source: project,
+    });
+
+    expect(stack).toHaveResourceLike('AWS::CodeStarNotifications::NotificationRule', {
+      DetailType: 'FULL',
+      Name: 'MyNotificationRule',
+      Resource: 'arn:aws:codebuild::1234567890:project/MyCodebuildProject',
+      Status: 'ENABLED',
+    });
+  });
+
+  test('created new notification rule with status DISABLED', () => {
+    const project = new FakeCodeBuild();
+
+    new notifications.NotificationRule(stack, 'MyNotificationRule', {
+      source: project,
+      enabled: false,
+    });
+
+    expect(stack).toHaveResourceLike('AWS::CodeStarNotifications::NotificationRule', {
+      Name: 'MyNotificationRule',
+      Resource: 'arn:aws:codebuild::1234567890:project/MyCodebuildProject',
+      Status: 'DISABLED',
+    });
+  });
+
+  test('created new notification rule with status ENABLED', () => {
+    const project = new FakeCodeBuild();
+
+    new notifications.NotificationRule(stack, 'MyNotificationRule', {
+      source: project,
+      enabled: true,
+    });
+
+    expect(stack).toHaveResourceLike('AWS::CodeStarNotifications::NotificationRule', {
+      Name: 'MyNotificationRule',
+      Resource: 'arn:aws:codebuild::1234567890:project/MyCodebuildProject',
+      Status: 'ENABLED',
+    });
+  });
+
+  test('notification added targets', () => {
+    const project = new FakeCodeBuild();
+    const topic1: notifications.INotificationRuleTarget = {
+      bind: () => ({
+        targetType: 'SNS',
+        targetAddress: 'arn:aws:sns::1234567890:MyTopic1',
+      }),
+    };
+    const topic2: notifications.INotificationRuleTarget = {
+      bind: () => ({
+        targetType: 'SNS',
+        targetAddress: 'arn:aws:sns::1234567890:MyTopic2',
+      }),
+    };
+    const topic3: notifications.INotificationRuleTarget = {
+      bind: () => ({
+        targetType: 'SNS',
+        targetAddress: 'arn:aws:sns::1234567890:MyTopic3',
+      }),
+    };
```

Comment:
Why are you not using `FakeSnsTopicTarget` for these?

Reply:
Already adjusted.